src/test/java/org/apache/commons/crypto/stream/AbstractCipherStreamTest.java

```diff
@@ -197,46 +224,246 @@ protected void doByteBufferRead(final String cipherClass, final boolean withChan
                 getCipher(cipherClass), smallBufferSize, iv, withChannel);
         buf.clear();
         byteBufferReadCheck(in, buf, 11);
+        in.close();      
+        
+        // Direct buffer, default buffer size, initial buffer position is 0, final read
```

Comment:
This says default but code says `smallBufferSize`. Not sure if that matters... also looking at the code you're adding, it seems a bit redundant with existing code, but didn't really double-check things.

Reply:
Nice catch.  The comment is a copy-paste oversight and I'll fix it.  I agree that these tests are largely redundant.  I needed to run different constructors and branch into some final logic in the read method to increase the code coverage.  I added a test that forced the read method to branch.  Then I just ran the tests again using a getCryptoInputSteam method that called different constructors.  It is a bit overkill, but I don't think it does any harm to run each test again against a separate constructor, as opposed to simply picking one test at random to cover the additional constructors.